### PR TITLE
fist step to adapt SOS work-flow to ttH work-flow

### PR DESCRIPTION
**Pull request by cbotta**
@peruzzim 
can you have a look? in the next step  I will implement all the CR in the sos_plots.py using mca_mcdata, then we discussed the preparation of the mca_frdata, and the inclusion of GIovanni's maps.

### CODE REVIEW

File: TTHAnalysis/python/plotter/functions.cc [hunk 1]
```diff
@@ -564,6 +564,13 @@ float puw2016_vtx(int nVtx) { if (nVtx<60) return _puw2016_vtx[nVtx]; else retur
...
 float puw2016_vtx_4fb(int nVtx) { if (nVtx<60) return _puw2016_vtx_4fb[nVtx]; else return 0; }
 
+
+// for up to 275125
+float _puw2016_vtx_4fb[60] = {1.0, 0.05100939489406209, 0.10753683801507614, 0.22479210497504293, 0.4061194548857097, 0.6424861757674111, 0.8699208198466813, 1.0886008554762319, 1.276892560891215, 1.4184863274330797, 1.4865216108423305, 1.5122068499515893, 1.4790079191970888, 1.4196901839857483, 1.3216638995044172, 1.2217703342251016, 1.0976458888260794, 0.9796814642869479, 0.8639064021650553, 0.7535695441176858, 0.6555461235682296, 0.5641253790572658, 0.4834647260351342, 0.4164426860491679, 0.35606001606142423, 0.30497669791734305, 0.2592054182625056, 0.22335323942907434, 0.19192071849647147, 0.1672517813653499, 0.14683806760557278, 0.13614259650259863, 0.11857021273678048, 0.11205636298564389, 0.10212521715621654, 0.08684353771752344, 0.08864604848027625, 0.09804676288437263, 0.09235565349796869, 0.07943403772193891, 0.14417507424675546, 0.08136302667712911, 0.05135680407279777, 0.1569931571656257, 0.12873438887581296, 0.3218359721895326, 0.1609179860947663, 0.8045899304738313, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0 };
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/functions.cc]: duplicate, please remove

File: TTHAnalysis/python/plotter/susy-sos/2los_plots.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+nJet40 : nJet40 : [-0.5,0.5,1.5,2.5,3.5,4.5,5.5,6.5] ; XTitle='N(jet\, p_{T} > 40)'
+ht:  htJet25-LepGood1_pt-LepGood2_pt :8,0.,400.; XTitle="Ht", Legend='TR'
+htJet25:   htJet25: 14,0,700  ; XTitle="H_{T}(jet p_{T} > 25)  [GeV]"
+htJet40j:   htJet40j: 14,0,700 ; XTitle="H_{T}(jet p_{T} > 40)  [GeV]"
+mhtJet25:   mhtJet25: 8,0,600  ; XTitle="H_{T}^{miss}(lep + jet p_{T} > 25)  [GeV]"
+mhtJet40:   mhtJet40: 8,0,600  ; XTitle="H_{T}^{miss}(lep + jet p_{T} > 40)  [GeV]"
+nBJet25L : nBJetLoose25  : [-0.5,0.5,1.5,2.5,3.5,4.5] ; XTitle='N(b-jet\, p_{T} > 25\, CSVL)'
+jet1Pt:  Jet1_pt: 12,0,600; XTitle="Leading jet p_{T}  [GeV]",Legend='TR'
+
+SR_bins_EWKino_100:SR_bins_EWKino_100(mass_2(LepGood1_pt,LepGood1_eta,LepGood1_phi,LepGood1_mass,LepGood2_pt,LepGood2_eta,LepGood2_phi,LepGood2_mass),met_pt,LepGood1_pdgId,LepGood2_pdgId):4,1,5;XTitle="M(ll)-flavour-bins",Legend='TR',IncludeOverflows=False, IncludeUnderflows=False, Legend='TR'
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/susy-sos/2los_plots.txt]: please remove special treatment of overflow/underflow here, it should be done always at the level of the cutfile

File: TTHAnalysis/python/plotter/susy-sos/2los_tight.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
+alwaystrue: 1
+trigger: Triggers_MET\b || Triggers_mmMET\b
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/susy-sos/2los_tight.txt]: without \b here, right?

File: TTHAnalysis/python/plotter/susy-sos/2los_triggerdefs.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Triggers_MET\b: HLT_MET ; AlsoData
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/susy-sos/2los_triggerdefs.txt]: it is safer to use explicit bits (HLT_BIT_HLT_...), as these can change with cfg configuration

File: TTHAnalysis/python/plotter/susy-sos/mca-2los-mc_76X.txt [hunk 5]
```diff
@@ -1,66 +1,89 @@
...
-T        : TToLeptons_sch :  xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0) ; Label="single t", FillColor=ROOT.kBlue-7, NormSystematic=0.5
-WZ 	 : WZTo1L3Nu: xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0);  Label="WZnp", FillColor=ROOT.kViolet-4, NormSystematic=0.3
-WZ 	 : WZTo1L1Nu2Q: xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0);  Label="WZnp", FillColor=ROOT.kViolet-4,  NormSystematic=0.3
-WW       : WWTo2L2Nu: xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0);  Label="WWnp", FillColor=ROOT.kAzure-8,  NormSystematic=0.3	
-WW       : WWToLNuQQ: xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0);  Label="WWnp", FillColor=ROOT.kAzure-8,  NormSystematic=0.3	
+WJets    : WJetsToLNu_LO :  xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0) && lheHT<=100; Label="W(fakes)", FillColor=ROOT.kCyan+2, NormSystematic=0.5
+WJets    : WJetsToLNu_HT100to200 : 0.25*xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0); Label="W(fakes)", FillColor=ROOT.kCyan+2, NormSystematic=0.5
+WJets    : WJetsToLNu_HT100to200_ext : 0.75*xsec: (LepGood1_mcMatchId==0 ||LepGood2_mcMatchId==0); Label="W(fakes)", FillColor=ROOT.kCyan+2, NormSystematic=0.5
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/susy-sos/mca-2los-mc_76X.txt]: new feature: one can avoid the 0.25, 0.75 by doing:

WJets : WJetsToLNu_HT100to200+WJetsToLNu_HT100to200_ext : xsec : ......

File: TTHAnalysis/python/plotter/susy-sos/mca-2los-mcdatacr.txt [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+#TTV : TTHnobb :  xsec: LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0; Label="Rares", FillColor=ROOT.kBlue+2, NormSystematic=0.5
+#Triboson : WZZ :  xsec: LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0; Label="Rares", FillColor=ROOT.kMagenta+2, NormSystematic=0.5
+#Triboson : WWZ :  xsec: LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0; Label="Rares", FillColor=ROOT.kMagenta+2, NormSystematic=0.5
+#Triboson : ZZZ :  xsec: LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0;Label="Rares", FillColor=ROOT.kMagenta+2, NormSystematic=0.5 
+#Rares : WpWpJJ : xsec : LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0 && LepGood3_mcMatchId!=0 ; FillColor=ROOT.kOrange-3
+#Rares : WWDouble : xsec : LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0 && LepGood3_mcMatchId!=0 ; FillColor=ROOT.kOrange-3
+
+
+
+data: SingleMuon_Run2016B_PromptReco_v2_runs_273150_274421
```
Comment (peruzzim) [inline TTHAnalysis/python/plotter/susy-sos/mca-2los-mcdatacr.txt]: Is this the only difference w.r.t. -mcdata.txt? If yes, there is no problem in running on using this as -mcdata.txt as triggers are exclusive in production. If one needs to speed up running, datasets can be excluded also matching on SingleMuon... or SingleElectron...

**Comment by cbotta** [comment]
@peruzzim 
I've taken into account all your comments, and I add some more stuff. Thanks for checking all this.

File: TTHAnalysis/python/plotter/susy-sos/mcc-lepWP.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+
+LepGood1_isTightCR\b : ((fabs(LepGood1_pdgId)==13 && LepGood1_softMuonId>0) || (fabs(LepGood1_pdgId)==11&&((LepGood1_pt<=10.0 && ((fabs(LepGood1_etaSc)<0.8 &&LepGood1_mvaIdSpring15>-0.265) || (fabs(LepGood1_etaSc)>=0.8 &&fabs(LepGood1_etaSc)<1.479&&LepGood1_mvaIdSpring15>-0.556)||(fabs(LepGood1_etaSc)>=1.479&& LepGood1_mvaIdSpring15 > -0.551))) || (LepGood1_pt>10.0 &&((fabs(LepGood1_etaSc)<0.8 &&LepGood1_mvaIdSpring15>0.87)||(fabs(LepGood1_etaSc)>=0.8 &&fabs(LepGood1_etaSc)<1.479 &&LepGood1_mvaIdSpring15>0.60)||(fabs(LepGood1_etaSc)>=1.479 &&LepGood1_mvaIdSpring15 >0.17)))))) && LepGood1_relIso03<0.5 && ((LepGood1_relIso03*LepGood1_pt)<5.||LepGood2_relIso03<0.1) && fabs(LepGood1_dxy)<0.03 && fabs(LepGood1_dz)<0.03 ; AlsoData
```
Comment (cbotta) [inline TTHAnalysis/python/plotter/susy-sos/mcc-lepWP.txt]: the other mca doesn't exist anymore. In order to have the TT defined in a way usable for the FR application as discussed yesterday I implemented the T cut in the main cut-flow

**Pull request merged**